Examples/DpdkBridge/main.cpp

```diff
@@ -0,0 +1,336 @@
+
+	int sendPacketsToPort = -1;
+
+	int optionIndex = 0;
+	char opt = 0;
+
+	uint32_t mBufPoolSize = DEFAULT_MBUF_POOL_SIZE;
+
+	while((opt = getopt_long (argc, argv, "d:c:m:hvl", FilterTrafficOptions, &optionIndex)) != -1)
+	{
+		switch (opt)
+		{
+			case 0:
+			{
+				break;
+			}
+			case 'd':
+			{
+				string portListAsString = string(optarg);
+				stringstream stream(portListAsString);
+				string portAsString;
+				int port;
+				// break comma-separated string into string list
+				while(getline(stream, portAsString, ','))
+				{
+					char c;
+					std::stringstream stream2(portAsString);
+					stream2 >> port;
+					if (stream2.fail() || stream2.get(c))
+					{
+						// not an integer
+						EXIT_WITH_ERROR_AND_PRINT_USAGE("DPDK ports list is invalid");
+					}
+					dpdkPortVec.push_back(port);
+				}
+				// verify list contains two ports
+				if(dpdkPortVec.size()!=2)
+				{
+					EXIT_WITH_ERROR_AND_PRINT_USAGE("DPDK list must contain two values");
+				}
+				break;
+			}
+			case 'c':
+			{
+				coreMaskToUse = atoi(optarg);
+				break;
+			}
+			case 'm':
+			{
+				mBufPoolSize = atoi(optarg);
+				break;
+			}
+			case 'h':
+			{
+				printUsage();
+				exit(0);
+			}
+			case 'v':
+			{
+				printAppVersion();
+				break;
+			}
+			case 'l':
+			{
+				listDpdkPorts();
+				exit(0);
+			}
+			default:
+			{
+				printUsage();
+				exit(0);
+			}
+		}
+	}
+
+	// verify list is not empty
+	if (dpdkPortVec.empty())
+	{
+		EXIT_WITH_ERROR_AND_PRINT_USAGE("DPDK ports list is empty. Please use the -d switch");
+	}
+
+	// extract core vector from core mask
+	vector<SystemCore> coresToUse;
+	createCoreVectorFromCoreMask(coreMaskToUse, coresToUse);
+
+	// need minimum of 2 cores to start - 1 management core + 1 (or more) worker thread(s)
+	if (coresToUse.size() < 2)
+	{
+		EXIT_WITH_ERROR("Needed minimum of 2 cores to start the application");
+	}
+
+	// initialize DPDK
+	if (!DpdkDeviceList::initDpdk(coreMaskToUse, mBufPoolSize))
+	{
+		EXIT_WITH_ERROR("Couldn't initialize DPDK");
+	}
+
+	// removing DPDK master core from core mask because DPDK worker threads cannot run on master core
+	coreMaskToUse = coreMaskToUse & ~(DpdkDeviceList::getInstance().getDpdkMasterCore().Mask);
+
+	// re-calculate cores to use after removing master core
+	coresToUse.clear();
+	createCoreVectorFromCoreMask(coreMaskToUse, coresToUse);
+
+	// collect the list of DPDK devices
+	vector<DpdkDevice*> dpdkDevicesToUse;
+	for (vector<int>::iterator iter = dpdkPortVec.begin(); iter != dpdkPortVec.end(); iter++)
+	{
+		DpdkDevice* dev = DpdkDeviceList::getInstance().getDeviceByPort(*iter);
+		if (dev == NULL)
+		{
+			EXIT_WITH_ERROR("DPDK device for port %d doesn't exist", *iter);
+		}
+		dpdkDevicesToUse.push_back(dev);
+	}
+
+	// go over all devices and open them
+	for (vector<DpdkDevice*>::iterator iter = dpdkDevicesToUse.begin(); iter != dpdkDevicesToUse.end(); iter++)
+	{
+		if (!(*iter)->openMultiQueues(1,1))
```

Comment:
Why limiting for 1 RX queue only? We can have another command-line parameter that indicates how many RX queues to open (default can be 1)

Reply:
I limited the example to just 1 RX and TX queue for simplicity. I have added support to specify multiple RX queues per device as requested (and also total available queues information on --list command). Still, all the RX queues are polled by the same worker thread, to avoid making the example more complex (I think the queue distribution logic in the FilterTraffic example will make it harder for someone new to PcapPlusPlus/DPDK to understand the code).